src/policies/activityReport.js

```diff
@@ -0,0 +1,69 @@
+/*
+  This class handles authorization for Activity Reports. In express handlers this
+  object should be constructed with the current user and the activity report to be
+  updated/retrieved for update/get or the new report for create. The permissions
+  of the user are checked against the report and simple questions about the actions
+  the user can take on that report are answered, mainly can the user create, update
+  or get the report.
+*/
+import _ from 'lodash';
+import SCOPES from '../middleware/scopeConstants';
+
+export default class ActivityReport {
+  constructor(user, activityReport) {
+    this.user = user;
+    this.activityReport = activityReport;
+  }
+
+  canCreate() {
+    return this.canWriteInRegion();
+  }
+
+  canUpdate() {
+    return (this.isAuthor() || this.isCollaborator()) && this.canWriteInRegion();
+  }
+
+  canGet() {
+    const { status } = this.activityReport;
+    const canReadUnapproved = this.isAuthor() || this.isCollaborator() || this.isApprovingManager();
+
+    if (canReadUnapproved) {
+      return canReadUnapproved;
+    }
+
+    if (status === 'approved') {
+      return this.canReadInRegion();
+    }
+
+    return false;
+  }
+
+  canWriteInRegion() {
+    const permissions = _.find(this.user.permissions,
+      (permission) => (
+        permission.scopeId === SCOPES.READ_WRITE_REPORTS
+        && permission.regionId === this.activityReport.regionId));
+    return !_.isUndefined(permissions);
+  }
+
+  canReadInRegion() {
+    const permissions = _.find(this.user.permissions,
+      (permission) => (
+        (permission.scopeId === SCOPES.READ_REPORTS
+        || permission.scopeId === SCOPES.READ_WRITE_REPORTS)
+        && permission.regionId === this.activityReport.regionId));
+    return !_.isUndefined(permissions);
+  }
+
+  isAuthor() {
+    return this.user.id === this.activityReport.userId;
+  }
+
+  isCollaborator() {
+    return this.activityReport.collaborators.some((user) => user.id === this.user.id);
+  }
+
+  isApprovingManager() {
+    return this.activityReport.approvingManagerId === this.user.id;
```

Comment:
For MVP there will only be one manager, but I wonder if designing for future expansion could be worth it at this point, e.g. include the manager in the collaborators table with a flag. That's just one way. 
But, if you think this would add a lot of extra work for now, we can postpone.

Reply:
That feels like something we can leave until after the MVP, especially because we aren't 100% sure how multiple managers are going to work at this point. I'd hate to start down a path now and have it turn out to be the wrong path. Should be easy to update this class once we add the ability for a report to have multiple managers.

Reply:
> Should be easy to update this class once we add the ability for a report to have multiple managers.
Once we are in prod we are going to have to deal with the data migration as well.